### DISCUSSION TITLE
I export the qso to a KLM file

### DISCUSSION THREAD

**Discussion by sp5smy** [magicbug/Cloudlog#2123] [General]
I export the qso to a KLM file and I get this message from the website
Regarding PHP error "Message: mkdir(): Permission denied":
What should I do to export the klm file

**Comment by AndreasK79** [comment]
Hi, it sounds like you need to set permissions to the klm folder in Cloudlog.

**Reply by AndreasK79** [reply]
I was a bit quick there. Cloudlog tries to make the klm folder, but does not have the rights to do so.

**Reply by sp5smy** [reply]
I changed the share and started working :)

**Comment by phl0** [comment]
The issue here is that the kml folder is to be created in the root directory of Cloudlog. So it is not covered by the chown commands recommended in the wiki.

@AndreasK79 What do you think about moving kml to a subfolder? For example uploads/? I think the ADIF exports are also created in there.

**Reply by AndreasK79** [reply]
Yeah, I don't see a problem with moving the folder.

**Reply by phl0** [reply]
Me too. If we ask the user for giving access to Cloudlog's root directory we would create a security issue because the webserver could access .git dir for example. That is why I vote for a separate sub-dir. And uploads/ should exist and is already covered in the Wiki re permissions.
